server/container_create_linux.go

```diff
@@ -948,13 +948,23 @@ func addOCIBindMounts(ctx context.Context, mountLabel string, containerConfig *t
 	}
 
 	if _, mountSys := mountSet["/sys"]; !mountSys {
-		m := rspec.Mount{
-			Destination: "/sys/fs/cgroup",
-			Type:        "cgroup",
-			Source:      "cgroup",
-			Options:     []string{"nosuid", "noexec", "nodev", "relatime", "ro"},
-		}
-		specgen.AddMount(m)
+    if node.CgroupIsV2() {
+      m := rspec.Mount{
+        Destination: "/sys/fs/cgroup",
+        Type:        "cgroup",
+        Source:      "cgroup",
+        Options:     []string{"nosuid", "noexec", "nodev", "relatime", "rw"},
+      }
+		  specgen.AddMount(m)
+    } else {
+      m := rspec.Mount{
+        Destination: "/sys/fs/cgroup",
+        Type:        "cgroup",
+        Source:      "cgroup",
+        Options:     []string{"nosuid", "noexec", "nodev", "relatime", "ro"},
+      }
```

Comment:
To reduce code duplication, let's do something like:
```suggestion
      m := rspec.Mount{
        Destination: "/sys/fs/cgroup",
        Type:        "cgroup",
        Source:      "cgroup",
        Options:     []string{"nosuid", "noexec", "nodev", "relatime"},
      }
      if node.CgroupIsV2() {
          m.Options = append(m.Options, "rw")
      } else {
          …
```